ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixClusterManager.java

```diff
@@ -156,9 +174,13 @@
             // received and handled.
             manager.addInstanceConfigChangeListener(clusterChangeHandler);
             logger.info("Registered instance config change listeners for Helix manager at {}", zkConnectStr);
+            manager.addIdealStateChangeListener(clusterChangeHandler);
+            logger.info("Registered ideal state change listeners for Helix manager at {}", zkConnectStr);
             // Now register listeners to get notified on live instance change in every datacenter.
             manager.addLiveInstanceChangeListener(clusterChangeHandler);
             logger.info("Registered live instance change listeners for Helix manager at {}", zkConnectStr);
+            routingTableProvider.addRoutingTableChangeListener(clusterChangeHandler, null);
+            logger.info("Registered routing table change listeners in {}", dcName);
```

Comment:
In the future, we may be able to remove `InstanceConfigChangeListener` and `LiveInstanceChangeListener` in this class because `RoutingTableProvider` already implements `ExternalViewChangeListener, InstanceConfigChangeListener, ConfigChangeListener, LiveInstanceChangeListener, CurrentStateChangeListener`